.github/workflows/lint.yml

```diff
@@ -0,0 +1,26 @@
+---
+# desc: lint sanity check
+name: lint
+
+on:
+  push:
+  workflow_dispatch:
+
+
+jobs:
+  lint:
+    name: pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5
+
+      - uses: terraform-linters/setup-tflint@19a52fbac37dacb22a09518e4ef6ee234f2d4987 # v4
+        name: Setup TFLint
+        with:
+          tflint_version: v0.50.3
+
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5
```

Comment:
same as above, would suggest to supply the version from `.tool-versions`
```
Warning: go-version input was not specified. The action will try to use pre-installed version.
go version go1.21.9 linux/amd64
```

even though you're using 1.22